src/Middleware/Spa/SpaProxy/src/build/Microsoft.AspNetCore.SpaProxy.targets

```diff
@@ -0,0 +1,26 @@
+﻿<Project>
+
+  <Target Name="WriteSpaConfigurationToDisk" BeforeTargets="AssignTargetPaths">
+    <PropertyGroup>
+      <_SpaProxyServerLaunchConfig>$(IntermediateOutputPath)spa.proxy.json</_SpaProxyServerLaunchConfig>
+      <_SpaRootFullPath>$([System.IO.Path]::Combine('$(MSBuildProjectDirectory)', '$(SpaRoot)').Replace('\','\\'))</_SpaRootFullPath>
+      <SpaProxyTimeoutInSeconds Condition="'$(SpaProxyTimeoutInSeconds)' == ''" >60</SpaProxyTimeoutInSeconds>
+    </PropertyGroup>
+    <ItemGroup>
+      <_SpaProxyServerLaunchConfigLines Include="{" />
+      <_SpaProxyServerLaunchConfigLines Include="  &quot;SpaProxyServer&quot;: {" />
+      <_SpaProxyServerLaunchConfigLines Include="    &quot;ServerUrl&quot;: &quot;$(SpaProxyServerUrl)&quot;," />
+      <_SpaProxyServerLaunchConfigLines Include="    &quot;LaunchCommand&quot;: &quot;$(SpaProxyLaunchCommand)&quot;," />
+      <_SpaProxyServerLaunchConfigLines Include="    &quot;WorkingDirectory&quot;: &quot;$(_SpaRootFullPath)&quot;," />
+      <_SpaProxyServerLaunchConfigLines Include="    &quot;MaxTimeoutInSeconds&quot;: &quot;$(SpaProxyTimeoutInSeconds)&quot;" />
+      <_SpaProxyServerLaunchConfigLines Include="  }" />
+      <_SpaProxyServerLaunchConfigLines Include="}" />
+    </ItemGroup>
+    <WriteLinesToFile File="$(_SpaProxyServerLaunchConfig)" Lines="@(_SpaProxyServerLaunchConfigLines)" WriteOnlyWhenDifferent="true" Overwrite="true" />
+    <ItemGroup>
+      <ContentWithTargetPath Include="$(_SpaProxyServerLaunchConfig)" CopyToOutputDirectory="PreserveNewest" CopyToPublishDirectory="Never" TargetPath="spa.proxy.json" />
+      <FileWrites Include="$(_SpaProxyServerLaunchConfig)" />
+    </ItemGroup>
+  </Target>
```

Comment:
This is an interesting design. TBH I was expecting developers to create and edit the `spa.proxy.json` file themselves - I didn't predict we'd be generating it at build time based on a set of MSBuild properties.

One drawback to generating it like this is with formatting. The logic here will produce an invalid JSON file if `SpaProxyLaunchCommand` contains a `"` character, which could actually be very common since people like to pass all kinds of arguments to their startup commands. Not sure if `\n` or other characters would cause issues too.

Do you think the MSBuild properties are more discoverable/convenient than just telling people to drop in a `spa.proxy.json` file to their project? In both cases they will only know about this by reading some docs or tutorial, so  in either design we're relying on them copy/pasting stuff.

Reply:
> One drawback to generating it like this is with formatting. The logic here will produce an invalid JSON file if `SpaProxyLaunchCommand` contains a `"` character, which could actually be very common since people like to pass all kinds of arguments to their startup commands. Not sure if `\n` or other characters would cause issues too.

In general you should put arguments inside `package.json` otherwise you loose the ability to run `npm/yarn start` from the cli. I started down that route where the command in the project file contained everything. There are alternatives to make this work for all people if needed, like wrapping the command in a script file and using the script for launch instead.